sympy/calculus/codomain.py

```diff
@@ -0,0 +1,166 @@
+from sympy import S, sympify, diff, limit, oo
+from sympy.calculus.singularities import singularities
+from sympy.sets.sets import Interval, Intersection, FiniteSet, Union, Complement, Set
+from sympy.solvers.solveset import solveset_real
+
+
+def codomain(func, domain, *syms):
+    """ Finds the range of a real-valued function, for a real-domain
+
+    Parameters
+    ==========
+
+    func: Expr
+          The expression whose range is to be found
+    domain: Union of Sets
+          The real-domain for the variable involved in function
+    syms: Tuple of symbols
+          Symbol whose domain is given
+
+    Raises
+    ======
+
+    NotImplementedError
+          The algorithms to find the range of the given function are
+          not yet implemented.
+    ValueError
+          The input is not valid.
+    RuntimeError
+          It is a bug, please report to the github issue tracker.
+
+    Examples
+    ========
+
+    >>> from sympy.calculus.codomain import codomain
```

Comment:
Since you have added `codomain` to `__init__.py`, just doing `from sympy.calculus import codomain` should do.